src/learning/InitializeProfile.cpp

```diff
@@ -0,0 +1,203 @@
+                                       AlternativesPerformance &altPerfs)
+    : conf(config), altPerformance_(altPerfs) {
+  conf.logger->debug("Starting ProfileInitializer object...");
+}
+
+ProfileInitializer::ProfileInitializer(Config &config,
+                                       const ProfileInitializer &profInit)
+    : conf(config), altPerformance_(profInit.getAlternativesPerformance()) {
+  conf.logger->debug("Starting ProfileInitializer object...");
+}
+
+AlternativesPerformance ProfileInitializer::getAlternativesPerformance() const {
+  return altPerformance_;
+}
+
+void ProfileInitializer::setAlternativesPerformance(
+    AlternativesPerformance &newAltPerfs) {
+  altPerformance_ = newAltPerfs;
+}
+
+std::vector<float> ProfileInitializer::categoryFrequency() {
+
+  // Extraction category rank from each alternative and storing it in values
+  std::unordered_map<std::string, Category> map =
+      altPerformance_.getAlternativesAssignments();
+  std::vector<int> values(map.size());
+
+  auto cat_value_selector = [](auto pair) {
+    return pair.second.getCategoryRank();
+  };
+
+  std::transform(map.begin(), map.end(), values.begin(), cat_value_selector);
+
+  // Counting the number of occurence for each category rank
+  std::unordered_map<int, int> counts;
+  for (auto v : values)
+    ++counts[v];
+
+  // Counting the number of alternatives in the altPerformance object
+  const int nbAlternatives = std::accumulate(
+      std::begin(counts), std::end(counts), 0,
+      [](int value, const std::unordered_map<int, int>::value_type &p) {
+        return value + p.second;
+      });
+
+  // Computing the frequency
+  std::vector<float> frequency;
+  for (auto const pair : counts) {
+    float c = pair.second;
+    frequency.push_back(c / nbAlternatives);
+  }
+  return frequency;
+}
+
+std::vector<std::string> ProfileInitializer::getProfilePerformanceCandidates(
+    const Criterion &crit, const Category &cat, const int nbCategories) {
+
+  int catAbove;
+  int catBelow;
+  std::vector<std::string> candidates;
+
+  // Category rank starts at 0
+  if (cat.getCategoryRank() == nbCategories - 1) {
+    catAbove = cat.getCategoryRank();
+    catBelow = catAbove - 1;
+  } else {
+    catBelow = cat.getCategoryRank();
+    catAbove = catBelow + 1;
+  }
+
+  for (auto pair : altPerformance_.getAlternativesAssignments()) {
+    if (pair.second.getCategoryRank() == catBelow ||
+        pair.second.getCategoryRank() == catAbove) {
+      candidates.push_back(pair.first);
+    }
+  }
+  return candidates;
+}
+
+float ProfileInitializer::weightedProbability(
+    const std::string altId, const Criterion &crit, const Category &catAbove,
+    const Category &catBelow, const int nbCategories,
+    const std::vector<float> &catFrequency, std::vector<std::string> candidates,
+    float delta) {
+  // creating imaginary profile performance for criterion crit
+  std::string critId = crit.getId();
+  float imaginaryProfilePerformance =
```

Comment:
In terms of optimization here it's not great, getPerf is in O(n_prof * n_crit). The Performance table should be itterated over directly to transform this operation in O(1). 

Consider changing the function to 
```
float ProfileInitializer::weightedProbability(
    &Perf perfs, const Category &catAbove,
    const Category &catBelow, const int nbCategories,
    const std::vector<float> &catFrequency, std::vector<std::string> candidates,
    float delta) {
```